cloudinit/sources/DataSourceOVF.py

```diff
@@ -684,4 +741,98 @@ def _raise_error_status(prefix, error, event, config_file, conf):
     util.del_dir(os.path.dirname(config_file))
     raise error
 
+
+def load_cloudinit_data(metaPath, userPath):
+    """
+    Load the cloud-init meta data, user data, cfg and network from the
+    given files
+    """
+    LOG.debug('load meta data from: %s: user data from: %s',
+              metaPath, userPath)
+    md = {}
+    cfg = {}
+    ud = None
+    network = None
+
+    md = load(util.load_file(metaPath).replace("\r", ""))
+
+    if 'network' in md:
+        network = md['network']
+        del md['network']
```

Comment:
Why remove the key from the metadata dictionary ?

Reply:
The network in md is saved into network, so we still need to keep it in md?

Reply:
@xiaofengw-vmware. cloud-init caches metadata is reads from the datasource and stores it in /var/lib/cloud/instance/obj.pkl as the cached datasource. the utility `cloud-init query ds` returns this datasource metadata JSON info to customers (and developers) who might need to debug the metadata content provided by the underlying datasource to see what is affecting the config on an instance. It is helpful to preserve as close to raw content as possible for better bug triage and analysis on the cloud-init deployed machines. If possible, please leave this key in the metadata in general. Otherwise on debug folks need to load the obj,pkl and inspect _network_config on the cached datasource instead of just `cloud-init query ds.meta_data.network`